### ISSUE TITLE
Make types in Microsoft.AspNetCore.DataProtection.KeyManagement.Internal internal

### ISSUE THREAD

**Issue by amcasey** [dotnet/aspnetcore#54712]
## Background and Motivation

IInternalXmlKeyManager and ICacheableKeyRingProvider are explicitly test hooks and the other types are implementation details. They aren't included in our public API documentation since they aren't intended for external consumption.

PR: https://github.com/dotnet/aspnetcore/pull/54707

## Proposed API

```diff
-namespace Microsoft.AspNetCore.DataProtection.KeyManagement.Internal;

-public sealed class CacheableKeyRing {}
-public struct DefaultKeyResolution {}
-public interface ICacheableKeyRingProvider {}
-public interface IDefaultKeyResolver {}
-public interface IInternalXmlKeyManager {}
-public interface IKeyRing {}
-public interface IKeyRingProvider {}
```

## Risks

There's very little reason to implement any of these interfaces, since the test hooks that accept them are internal.  There's very little reason to invoke them, since the members of the returned objects are internal.  It is, however, not impossible that people are using them.  Furthermore, **I don't actually know why they were public in the first place**.

**Comment by Tratcher** [comment]
How does this fit with the API Compat policy of "Never remove/break a public API, and put back anything we removed in the past."?

**Comment by amcasey** [comment]
> How does this fit with the API Compat policy of "Never remove/break a public API, and put back anything we removed in the past."?

I guess I'm struggling to see the harm in removing an undocumented type you can't instantiate, subtype, or access the members of.  But, yes, it would technically be a violation of our ironclad guarantee against breaking changes.

**Comment by Tratcher** [comment]
@joperezr

**Comment by amcasey** [comment]
Does our compat guarantee require that they continue to function as they do now or just that they continue to exist?  If I made copies in Microsoft.AspNetCore.DataProtection.KeyManagement (no ".Internal") and left these orphaned, would that suffice?

Motivating example: I [introduced](https://github.com/dotnet/aspnetcore/pull/54517/commits/a1ef3f9fad054a96fb2c90ec7c378e692dd007f9#diff-f1f33c73258770ed52bba37d1debe5ac1b9fe61430006a6895ebbbd0bd3cff35R338) a new interface to avoid breaking the existing `ICacheableKeyRingProvider` - was it important to continue to implement the old test hook in case someone was calling it or could I have just dropped that the implemented interfaces list?  (Note that that's not the final shape of the [PR](https://github.com/dotnet/aspnetcore/pull/54517).)
